app/gui/view/graph-editor/src/component/node/input/widget.rs

```diff
@@ -8,54 +8,181 @@ use ensogl::data::color;
 use ensogl::display;
 use ensogl::display::object::event;
 use ensogl_component::drop_down::Dropdown;
-use frp::stream::EventEmitter;
 
 
 
-// ==================
-// === NodeWidget ===
-// ==================
+/// =================
+/// === Constants ===
+/// =================
+
+const DOT_COLOR: color::Lch = color::Lch::new(0.56708, 0.23249, 0.71372);
+
+
+
+// ===========
+// === FRP ===
+// ===========
 
 ensogl::define_endpoints_2! {
     Input {
-        set_current_value(Option<ImString>),
-        set_focused(bool),
+        set_metadata   (Option<Metadata>),
+        set_node_data  (NodeData),
+        set_current_value (Option<ImString>),
+        set_focused       (bool),
+        set_visible       (bool),
     }
     Output {
         value_changed(Option<ImString>),
     }
 }
 
-/// Possible widgets for a node input.
-///
-/// Currently all widget types are hardcoded. This is likely to be a temporary solution. In the
-/// future the set of widget types might be dynamic, similar to visualizations.
-#[derive(Clone, Debug, CloneRef)]
-pub enum Widget {
-    /// A widget for selecting a single value from a list of available options.
-    SingleChoice(SingleChoice),
+/// Widget metadata that needs to be fetched from the graph asynchronously.
+#[derive(Debug, Clone)]
+pub struct Metadata {
+    /// The kind of widget to use.
+    pub kind:            Kind,
+    /// The widget display mode.
+    pub display:         Display,
+    /// Dynamically fetched widget entries.
+    pub dynamic_entries: Vec<ImString>,
```

Comment:
IMO these docs are not needed - they make code only longer to read. If we have field "kind" in a struct `widget::Metadata`, then its aboviouss its a widget kind. If you feel that the name does not fully tell what it is, change the name instead to make it more descriptive - other places in the code will benefit from this change.